litestar/_asgi/asgi_router.py

```diff
@@ -158,27 +158,28 @@ async def lifespan(self, receive: LifeSpanReceive, send: LifeSpanSend) -> None:
         Returns:
             None.
         """
-
-        message = await receive()
         shutdown_event: LifeSpanShutdownCompleteEvent = {"type": "lifespan.shutdown.complete"}
         startup_event: LifeSpanStartupCompleteEvent = {"type": "lifespan.startup.complete"}
 
+        await receive()
+
+        started = False
         try:
             async with self.app.lifespan():
                 await send(startup_event)
-                message = await receive()
+                started = True
+                await receive()
```

Comment:
The issue being that if the exception comes from the lifespan context before a shutdown message is received but after we've confirmed the startup, then `message` is still the startup one causing us to send the startup failure in the exception handler.